contents/tutorials/nextjs-surveys.md

```diff
@@ -0,0 +1,536 @@
+
+```JSON
+[
+   {
+      "id":"018ad0e0-0de6-0000-6a56-033975bd0c68",
+      "name":"my-first-survey",
+      "description":"",
+      "type":"api",
+      "questions":[
+         {
+            "type":"rating",
+            "scale":10,
+            "display":"number",
+            "question":"How likely are you to recommend us to a friend?",
+            "description":"",
+            "lowerBoundLabel":"Unlikely",
+            "upperBoundLabel":"Very likely"
+         }
+      ],
+      "conditions":null,
+      "appearance":{
+         "textColor":"black",
+         "whiteLabel":false,
+         "backgroundColor":"white",
+         "submitButtonText":"Submit",
+         "ratingButtonColor":"#e0e2e8",
+         "submitButtonColor":"#2c2c2c",
+         "descriptionTextColor":"#4b4b52",
+         "thankYouMessageHeader":"Thank you for your feedback!",
+         "displayThankYouMessage":true
+      },
+      "start_date":"2023-09-26T09:44:31.844000Z",
+      "end_date":null
+   }
+]
+```
+
+We can use this survey object to configure our `Survey` component:
+
+```react
+  // ... rest of your code ...
+
+  const [surveyTitle, setSurveyTitle] = useState(false);
+  const [surveyID, setSurveyID] = useState(false);
+
+  useEffect(() => {
+    posthog.getActiveMatchingSurveys((surveys) => {
+      if (surveys.length > 0) {
+        const survey = surveys[0];
+        setSurveyID(survey.id);
+        setSurveyTitle(survey.questions[0].question)
+      }
+    }); 
+  }, [posthog])
+  
+  // ... rest of your code ...
+
+  return (
+    <main className={styles.main}>
+      <div className="App">
+        <h1>This is our Next.js survey tutorial</h1>
+        {showSurvey && (
+        <Survey
+          title={surveyTitle}
+          onDismiss={handleDismiss}
+          onSubmit={handleSubmit}
+        />)}
+      </div>
+    </main>
+  )
+```
+
+Finally, we want to make sure we don't show the survey again to users who have either submitted or dismissed it. 
+
+We use [`localStorage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) to store this data. Then, we'll add a check to show the survey based on whether the user has already interacted with it or not:
+
+```react
+  // ... rest of your code ...
+
+  const [showSurvey, setShowSurvey] = useState(true);
+
+  useEffect(() => {
+    // Check local storage to see if the user has already seen this particular survey
+    const hasInteractedWithSurvey = localStorage.getItem(`hasInteractedWithSurvey_${surveyID}`);
+    setShowSurvey(!hasInteractedWithSurvey);
+  }, [surveyID]);
+
+  const handleDismiss = () => {
+    setShowSurvey(false);
+    console.log("Survey dismissed!");
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');
+  };
+
+  const handleSubmit = (value) => {
+    setShowSurvey(false);
+    console.log("User submitted:", value);
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');  
+  };
```

Comment:
1. Could probably combine the setShowSurvey and localStorage set item logic so it doesn't repeat
2. Can get rid of the console.log at this point

Reply:
> Could probably combine the setShowSurvey and localStorage set item logic so it doesn't repeat

Yeah, if this was real code I'd do that to avoid repetition, but I'll leave as is as I think it's either to follow this way